### ISSUE TITLE
Questions about the training setup

### ISSUE THREAD

**Issue by 2toinf** [intuitive-robots/flower_vla_calvin#1]
Thanks for open-sourcing this great work! I have a couple of questions while trying to reproduce the Calvin results after reading your paper and code: 
1. The paper states that Flower's Calvin experiments didn't use pre-training. However, the config shows:
```
load_pretrained: True
pretrained_model_path: /home/hk-project-sustainebot/ft4740/code/flower_vla_policy/logs/runs/2025-02-05/10-17-02/360000_model_weights.pt
```
This is confusing. Is there a potential issue with the prompt or configuration here?  We also found it very difficult to train a good model with only 40k gradient steps and a batch size of 32 when trying to reproduce Flower ourselves.
2. The paper mentions using a "Custom LR," but I can't find where to set it in the code and don't know how to configure it. Could you please provide some guidance on how to set up this custom learning rate?
Thanks a lot!

**Comment by mbreuss** [comment]
Hi,

thanks for the interest in our repo and model.

> The paper states that Flower's Calvin experiments didn't use pre-training.

Sorry, I am working on an updated paper version, where I use both variants and to achieve the best results the pertaining helps a bit. Thus, I added the finetning code here. The pretrained model_path can be downloaded from HF see ink below. If you are interested in training from scratch, I can add the dual optimizer version some time this week too. 
Here is the pretrained checkpoint: 

https://huggingface.co/mbreuss/flower_vla_pret/blob/main/360000_model_weights.pt  

Just download it and replace this path with the your local one.

> We also found it very difficult to train a good model with only 40k gradient steps and a batch size of 32 when trying to reproduce Flower ourselves

That is weird. Did you use 4 GPUs and what did you change for the training setup? Can you try it with the pretrained checkpoint again? Also for a batch size of 32 you need to keep the batch size parameter in the main config at 8 since it refers to individual GPU one. 

For reference here is a training curve from a random ABC on how the trainings look like:

![Image](https://github.com/user-attachments/assets/e32b66a3-6d53-4b5d-a231-49529bbaa522)


> The paper mentions using a "Custom LR," but I can't find where to set it in the code and don't know how to configure it.

This has been used for pretraining the model, since this repo right now only contains the finetuning part, it works with a single shared LR and optimizer. Otherwise the custom LR was just to use two separate optimizers with a lr of 1e-4 for the Flow part and 1e-5 for the VLM weights. Again if you are interested in that, I can upload the from scratch version too.

Let me know if that helps and feel free to reach out again!

**Comment by Holmes-GU** [comment]
> Thanks for open-sourcing this great work! I have a couple of questions while trying to reproduce the Calvin results after reading your paper and code:
> 
> 1. The paper states that Flower's Calvin experiments didn't use pre-training. However, the config shows:
> 
> ```
> load_pretrained: True
> pretrained_model_path: /home/hk-project-sustainebot/ft4740/code/flower_vla_policy/logs/runs/2025-02-05/10-17-02/360000_model_weights.pt
> ```
> 
> This is confusing. Is there a potential issue with the prompt or configuration here? We also found it very difficult to train a good model with only 40k gradient steps and a batch size of 32 when trying to reproduce Flower ourselves. 2. The paper mentions using a "Custom LR," but I can't find where to set it in the code and don't know how to configure it. Could you please provide some guidance on how to set up this custom learning rate? Thanks a lot!

Hi,

Would you like to share the paper? I could not find it from both arXiv and Google Scholar. Thanks

**Comment by mbreuss** [comment]
its only a workshop paper for now but I am working on an arXiv release:

https://openreview.net/pdf?id=ifo8oWSLSq

**Comment by mbreuss** [comment]
During pretraining I only use the static image from all datasets. Most datasets of OXE have only 1 image anyways. For finetuning I then extend it to the amount of required images. I already tried pretraining with up to 3 images but it makes everything much more complicated and memory intensive and it does not seem to benefit downstream tuning.
Each dataset actions are normalized betwen -1 and 1 following the octo style and code.

**Comment by Holmes-GU** [comment]
> During pretraining I only use the static image from all datasets. Most datasets of OXE have only 1 image anyways. For finetuning I then extend it to the amount of required images. I already tried pretraining with up to 3 images but it makes everything much more complicated and memory intensive and it does not seem to benefit downstream tuning. Each dataset actions are normalized betwen -1 and 1 following the octo style and code.

Copy that. When I try to run the vla flower, I encounter the following issues. Would you like to give suggestions about this? Thanks! 

<img width="1413" alt="Image" src="https://github.com/user-attachments/assets/2c5ebb46-cb6a-4a24-b5b4-b58db3a99230" />

**Comment by mbreuss** [comment]
Try this:

<img width="851" alt="Image" src="https://github.com/user-attachments/assets/5ae313d1-0bc2-4596-9440-143886475b96" />

I put it in the readme but maybe need to highlight it more

**Comment by Holmes-GU** [comment]
> Try this:
> 
> <img alt="Image" width="851" src="https://private-user-images.githubusercontent.com/33622678/434885588-5ae313d1-0bc2-4596-9440-143886475b96.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDQ5MDY0MTUsIm5iZiI6MTc0NDkwNjExNSwicGF0aCI6Ii8zMzYyMjY3OC80MzQ4ODU1ODgtNWFlMzEzZDEtMGJjMi00NTk2LTk0NDAtMTQzODg2NDc1Yjk2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA0MTclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNDE3VDE2MDgzNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWFjMjBjZDJhNmNjMWM5MjFlYzZhZTRhNjRmYmU2OTg4MTEzOTVmYTRkNGM3NTZmNTc1NWFiMTgyMzUwNjZiZjkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.ktXM5pLjknmhWLQeKdUHf6SR_fhc39YCJ4FD1IfC10A">
> I put it in the readme but maybe need to highlight it more

Thanks for your advice. Would like to help find the position of "log.info(f"Using calvin_env with commit {get_git_commit_hash(Path(calvin_env.__file__))}.")"?

**Comment by mbreuss** [comment]
https://github.com/mees/calvin_env/blob/797142c588c21e76717268b7b430958dbd13bf48/calvin_env/envs/play_table_env.py#L72

**Comment by Holmes-GU** [comment]
> https://github.com/mees/calvin_env/blob/797142c588c21e76717268b7b430958dbd13bf48/calvin_env/envs/play_table_env.py#L72

Thanks. I have successfully reproduced the reported results on the CALVIN. Looking forward to reproducing results via OXE pretraining and LIBERO.
